Fixed #1035, #1048, #1034

yukozh opened the pull request:
/cc @glennc @Tratcher @Eilon 

In this PR, I have fixed:
- Fixed #1035 Remove web listener
- Simple the command line of kestrel
- Add IISPlatformHandler
- Fixed #1048 Use the correct System.Console
- Fixed #1034 Up the latest to date
- Created rc1 samples
- Fixed docker files(COPY file problems)
- Rename NuGet.Config into NuGet.config

dnfclas commented:
Hi __@Kagamine__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

Kagamine commented:
Cloud I create a folder to contian the rc1 samples?

Kagamine commented:
@glennc @pranavkm Can you review this?

davidfowl commented on samples/1.0.0-beta8/HelloMvc/project.json:
```diff
@@ -11,14 +11,14 @@
     "commands": {
-        "web": "Microsoft.AspNet.Hosting --server Microsoft.AspNet.Server.WebListener --server.urls http://localhost:5001",
-        "kestrel": "Microsoft.AspNet.Hosting --server Microsoft.AspNet.Server.Kestrel --server.urls http://*:5004"
+        "web": "Microsoft.AspNet.Server.Kestrel --server.urls http://*:5004"
```
I'd prefer localhost here

Kagamine replied:
OK

davidfowl commented:
Remove the View from hello mvc

Kagamine commented:
Okay

glennc commented on samples/1.0.0-rc1/HelloMvc/Dockerfile:
```diff
@@ -0,0 +1,38 @@
+FROM ubuntu:14.04
```
The rc1 Dockerfile should assume the existence of the Microsoft rc1 image. So just from Microsoft/aspnet:1.0.0-rc1, copy app, restore, build. Like the others.

Kagamine replied:
All right

Kagamine commented:
/cc @glennc done

dnfclas commented:
Hi __@Kagamine__, I'm your friendly neighborhood .NET Foundation Pull Request Bot (You can call me DNFBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <p>The agreement was validated by .NET Foundation and real humans are currently evaluating your PR.</p>

TTYL, DNFBOT;

glennc commented on samples/1.0.0-rc1/HelloWeb/Dockerfile:
```diff
@@ -0,0 +1,9 @@
+FROM microsoft/aspnet:1.0.0-rc1
+
+COPY project.json /app/
```
Change this to just copy the app.

COPY . /app./

and then remove the other copy line. Since beta8 shipped doing this two-stage copy can introduce subtle bugs.

glennc commented:
Outside of that last comment this looks reasonable enough to merge. It will need to wait until other RC stuff is pushed, the rc1 docker image for example, but I can do that as everything gets pushed.

Kagamine commented:
/cc @glennc  done

glennc commented:
I have a repo that we add the Dockerfiles to here: https://github.com/aspnet/aspnet-win-docker

But uploading them isn't as easy, since the images for Windows aren't on a normal Dockerhub repo.

One other thing, can you add a NuGet.config entry to point to https://www.myget.org/F/aspnetrc1/api/v2 and then test out the RC1 samples? You will need to be careful about cleaning all the packages that are restored off your machine. But it will prove they work. You will need to change all the version numbers in the project.json to be rc1-final not just rc1. As soon as you've done that and made any changes that are needed I will merge this.

The rc1 Docker image will be called rc1-final as well.

Kagamine commented:
But, when the rc1 released to nuget, we still have to change it into rc1.

Kagamine commented:
In this repo, the samples are all using the stable feed. yea?

Kagamine commented:
which repo can get the windows container?

glennc commented:
The packages released to NuGet are going to be rc1-final, and so is the Docker image. I'm not sure how the Windows container is pulled onto windows machines, there is some special mechanism. I will be creating an RC1 windows container later tonight, but I will not be able to push it anywhere that anyone else can pull right now.

glennc commented:
the aspnetrc1 MyGet feed has the candidate RC1 builds with their final version numbers. We have pushed to it several times, so the actual binaries can change on the pre-release feeds. but the package versions and such are how they are going to be.

Kagamine commented:
/cc @glennc  done

yukozh committed:
Fixed #1035, #1048, #1034. Add rc1 samples, update dockerfile

Remove web listener
Simple the command line of kestrel
Add IISPlatformHandler
Up the latest to date
Created rc1 folder
Update docker files
rc1 -> rc1-final

glennc commented:
Thanks @Kagamine I appreciate all the effort.

Kagamine commented:
@glennc I think #1035 should change the label into 3-Done

glennc commented:
Changed, nice to find someone noticing the details and pinging us on them :)
